File: baekjoon/js/10809.js
const fs = require("fs");
const buf = fs.readFileSync(0, "utf-8").toString().trim();

const result = Array(26).fill(-1);

for (let i = 0; i < buf.length; i++) {
  const code = buf.charCodeAt(i) - 97;

  if (result[code] === -1) {
    result[code] = i;
  }
}

console.log(result.join(" "));

File: baekjoon/rust/10809.rs
use std::io::stdin;

fn main() {
    let mut buf = String::new();
    stdin().read_line(&mut buf).unwrap();

    let mut alphabet = [-1; 26];

    for (i, b) in buf.trim().bytes().enumerate() {
        let code = (b - 97) as usize;

        if alphabet[code] == -1 {
            alphabet[code] = i as i32;
        }
    }

    let result = alphabet
        .iter()
        .map(|&x| x.to_string())
        .collect::<Vec<String>>();

    print!("{}", result.join(" "));
}
